tests/functional/codegen/features/decorators/test_pure.py

```diff
@@ -49,58 +47,120 @@ def foo() -> int128: pure
 def foo(a: address) -> int128:
     return staticcall Foo(a).foo()
     """
-    )
+    c1 = get_contract(code1)
+    c2 = get_contract(code2)
     assert c2.foo(c1.address) == 5
 
 
-def test_invalid_envar_access(get_contract, assert_compile_failed):
-    assert_compile_failed(
-        lambda: get_contract(
-            """
+def test_invalid_envar_access(get_contract):
+    code = """
 @pure
 @external
 def foo() -> uint256:
     return chain.id
     """
-        ),
-        StateAccessViolation,
-    )
+    with pytest.raises(StateAccessViolation):
+        compile_code(code)
+
+
+def test_invalid_codesize_access(get_contract):
+    code = """
+@pure
+@external
+def foo() -> uint256:
+    return msg.sender.codesize
```

Comment:
`msg.sender` is itself blocked, so we don't know whether it fails to compile for the right reason?

Reply:
nice find. fixed in https://github.com/vyperlang/vyper/pull/3895/commits/45192f67f3a596129a09d1d086e07d99cec54c17